electron/execution.ts

```diff
@@ -155,7 +163,7 @@ async function onTest(data, browserWindow) {
   };
 
   // returns TestEvent interface defined in common/types.ts
-  const constructEvent = parsed => {
+  const constructEvent: (parsed: Record<string, any>) => TestEvent | undefined = parsed => {
```

Comment:
```suggestion
  const constructEvent: (parsed: Record<string, unknown>) => TestEvent | undefined = parsed => {
```

I'd rather see us employ `unknown` here. We can define a [type guard](https://www.typescriptlang.org/docs/handbook/advanced-types.html#user-defined-type-guards) to check for the appropriate fields, or change the `switch` block to an `if...else` chain that can check for the appropriate fields that provides the correct type annotation.

Reply:
The type guard suggestion sounds fantastic 🌟 
However, IMHO, I'm not sure using `unknown` here would bring a big benefit here just because we need to cast the type of `parsed` one more time just before constructing a structured data from it(it seems to me like we will need to define a type for the data that lasts only temporarily).  I thought maybe we keep the parsed type as `Record<string, any>` and add the type checking with type guard on constructed data so that they 100% meet the type definition of `TestEvent`? WDYT?

Reply:
Sure, we can do that. The reason I preferred  `unknown` over `any` is that `unknown` will automatically flag any usage of potentially missing fields. But if we're going to expose all the code paths here to type guards anyway I suppose that's also an acceptable solution.

Reply:
I've added `if..else`s before returning the TestEvent objects as you suggested: https://github.com/elastic/synthetics-recorder/pull/205/commits/973d4a544647707e47d05f1167ef810cacf4366b